api/src/main/java/ca/bc/gov/educ/api/grad/report/service/SchoolReportsService.java

```diff
@@ -0,0 +1,132 @@
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class SchoolReportsService extends BaseService {
+  SchoolReportsRepository schoolReportsRepository;
+  SchoolReportsLightRepository schoolReportsLightRepository;
+  SchoolReportsTransformer schoolReportsTransformer;
+  CommonService commonService;
+
+  private static final Logger logger = LoggerFactory.getLogger(SchoolReportsService.class);
+
+  @Autowired
+  public SchoolReportsService(SchoolReportsRepository schoolReportsRepository, SchoolReportsTransformer schoolReportsTransformer, CommonService commonService, SchoolReportsLightRepository schoolReportsLightRepository) {
+    this.schoolReportsRepository = schoolReportsRepository;
+    this.schoolReportsTransformer = schoolReportsTransformer;
+    this.commonService = commonService;
+    this.schoolReportsLightRepository = schoolReportsLightRepository;
+  }
+
+  private <T> Specification<T> buildSearchSpecification(UUID schoolOfRecordId, String reportTypeCode) {
+    Specification<T> spec = Specification.where(null);
+
+    if (schoolOfRecordId != null) {
+      spec = spec.and(SchoolReportsSpecifications.hasSchoolOfRecordId(schoolOfRecordId));
+    }
+    if (StringUtils.isNotBlank(reportTypeCode)) {
+      spec = spec.and(SchoolReportsSpecifications.hasReportTypeCode(reportTypeCode));
+    }
+    return spec;
+  }
+
+  public List<SchoolReports> searchSchoolReports(UUID schoolOfRecordId, String reportTypeCode, boolean isLight) {
+    if(isLight) {
+      Specification<SchoolReportsLightEntity> spec = buildSearchSpecification(schoolOfRecordId, reportTypeCode);
+      List<SchoolReportsLightEntity> entities = schoolReportsLightRepository.findAll(spec);
+      return schoolReportsTransformer.transformToLightDTO(entities);
+    } else {
+      Specification<SchoolReportsEntity> spec = buildSearchSpecification(schoolOfRecordId, reportTypeCode);
+      List<SchoolReportsEntity> entities = schoolReportsRepository.findAll(spec);
+      return schoolReportsTransformer.transformToDTO(entities);
+    }
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public ResponseEntity<InputStreamResource> getSchoolReportBySchoolOfRecordIdAndReportType(UUID schoolOfRecordId, String reportTypeCode) {
+    SchoolReports schoolReport = schoolReportsTransformer.transformToDTO(schoolReportsRepository.findBySchoolOfRecordIdAndReportTypeCode(schoolOfRecordId, reportTypeCode));
+    if (schoolReport != null && schoolReport.getReport() != null) {
+      byte[] reportByte = Base64.decodeBase64(schoolReport.getReport().getBytes(StandardCharsets.US_ASCII));
+      ByteArrayInputStream bis = new ByteArrayInputStream(reportByte);
+      return ResponseEntity
+              .ok()
+              .contentType(MediaType.APPLICATION_PDF)
+              .body(new InputStreamResource(bis));
```

Comment:
please move ResponseEntity up to controller level

Reply:
updated